### PR TITLE
Create single TableReference instance per table

### PR DESCRIPTION
**Pull request by emolsson**
Also make sure that tables exists before creating them.

This is in preparation in order to be able to use UUID to
represent a table.

### CODE REVIEW

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReference.java [hunk 1]
```diff
@@ -14,55 +14,31 @@
...
-    @Override
-    public int hashCode()
-    {
-        int result = myKeyspace != null ? myKeyspace.hashCode() : 0;
-        result = 31 * result + (myTable != null ? myTable.hashCode() : 0);
-        return result;
-    }
+    /**
+     * Get the table id.
+     *
```
Comment (itskarlsson) [inline core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReference.java:26]: Nit: Do we really need these descriptions? I suggest we either remove the description and leave the return or we remove it all together. I think most people can probably figure out what the function does just by the name of it.

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReferenceFactory.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.core.utils;
+
+/**
+ * A table reference factory to generate table references.
```
Comment (itskarlsson) [inline core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReferenceFactory.java:18]: A factory that generates table references based on keyspace and table name

File: CHANGES.md [hunk 3]
```diff
@@ -19,6 +19,7 @@
 * Use yaml format for configuration - Issue #126
 * Use springboot for REST server - Issue #111
 * Add Health Endpoint - Issue #131
+* Use table id in metric names
```
[the remark this answers is not in the document]
Comment (emolsson) [reply CHANGES.md:22]: Yeah, it's part of #120 so I'll add that.

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReference.java (hunk 1 above)
Comment (emolsson) [reply core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReference.java:26]: Done

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReferenceFactory.java (hunk 2 above)
Comment (emolsson) [reply core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/TableReferenceFactory.java:18]: Done

**Review by itskarlsson — APPROVED** [review]

**Pull request merged**